server/src/main/java/io/crate/planner/operators/JoinPlan.java

```diff
@@ -46,6 +46,7 @@ public class JoinPlan extends AbstractJoinPlan {
     private final boolean rewriteFilterOnOuterJoinToInnerJoinDone;
     private final boolean lookUpJoinRuleApplied;
     private final boolean moveConstantJoinConditionRuleApplied;
+    private final boolean eliminateCrossJoinRuleIsApplied;
```

Comment:
The cross-join elimination should be only run once.

Reply:
is it only a matter of not needed work, or can it lead to wrong plans if try to apply it more than once?

Reply:
Reapplication can not lead to wrong plans. If the rule matches again, no improvements will be made. Then the rule returns `null`, and the optimiser will skip the rule. But this can all be avoided. So this is just to safe cycles.

Reply:
thx!